src/main/java/com/box/sdk/BoxWebHook.java

```diff
@@ -0,0 +1,687 @@
+     * {@link BoxWebHook.TargetType}.
+     *
+     * @param targetType
+     *            on which target the triggers should be applied to
+     * @param triggers
+     *            for check
+     *
+     * @see #validateTrigger(TargetType, Trigger)
+     */
+    public static void validateTriggers(BoxWebHook.TargetType targetType, Collection<BoxWebHook.Trigger> triggers) {
+        for (BoxWebHook.Trigger trigger : triggers) {
+            validateTrigger(targetType, trigger);
+        }
+    }
+
+    /**
+     * Validates that provided {@link BoxWebHook.Trigger} can be applied on the provided {@link BoxWebHook.TargetType}.
+     *
+     * @param targetType
+     *            on which targets the trigger should be applied to
+     * @param trigger
+     *            for check
+     *
+     * @see #validateTriggers(TargetType, Collection)
+     */
+    private static void validateTrigger(BoxWebHook.TargetType targetType, BoxWebHook.Trigger trigger) {
+        for (BoxWebHook.TargetType type : trigger.getTypes()) {
+            if (targetType.equals(type)) {
+                return;
+            }
+        }
+        throw new IllegalArgumentException(String.format(
+                "Provided trigger '%s' is not supported on provided target '%s'.", trigger.name(), targetType.name()));
+    }
+
+    /**
+     * @return Gets information about this {@link BoxWebHook}.
+     */
+    public BoxWebHook.Info getInfo() {
+        URL url = WEBHOOK_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        BoxAPIRequest request = new BoxAPIRequest(this.getAPI(), url, HttpMethod.GET);
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        return new Info(JsonObject.readFrom(response.getJSON()));
+    }
+
+    /**
+     * Updates {@link BoxWebHook} information.
+     *
+     * @param info
+     *            new state
+     */
+    public void updateInfo(BoxWebHook.Info info) {
+        URL url = WEBHOOK_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        BoxJSONRequest request = new BoxJSONRequest(getAPI(), url, HttpMethod.PUT);
+        request.setBody(info.getPendingChanges());
+
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject jsonObject = JsonObject.readFrom(response.getJSON());
+        info.update(jsonObject);
+    }
+
+    /**
+     * Deletes this webhook.
+     */
+    public void delete() {
+        URL url = WEBHOOK_URL_TEMPLATE.build(getAPI().getBaseURL(), this.getID());
+        BoxAPIRequest request = new BoxAPIRequest(getAPI(), url, HttpMethod.DELETE);
+        BoxAPIResponse response = request.send();
+        response.disconnect();
+    }
+
+    /**
+     * Contains information for a {@link BoxWebHook} instance.
+     *
+     * @author Stanislav Dvorscak
+     *
+     */
+    public class Info extends BoxResource.Info {
+
+        /**
+         * @see #getTarget()
+         */
+        private Target target;
```

Comment:
@stanislav-dvorscak: For me adding a new method to BoxResource seems to be a better choice. This would be backward compatible and help get rid of Target and TargetType. However, I will let @gcurtis or @tarrencev confirm that.